packages/realm_dart/test/realm_logger_test.dart

```diff
@@ -1,211 +1,105 @@
+    Realm.logger.setLogLevel(RealmLogLevel.off);
+    Realm.logger.setLogLevel(RealmLogLevel.all, category: RealmLogCategory.realm.sdk);
 
-        Future<void> logTestMessage(Level level, String message) async {
-          completer2 = Completer<void>();
-          RealmInternal.logMessageForTesting(level, message);
-          if (Realm.logger.level == RealmLogLevel.off) {
-            Future<void>.delayed(const Duration(milliseconds: 10)).then((value) {
-              completer2.complete();
-            }).ignore();
-          }
-          await completer2.future;
-        }
-
-        await logTestMessage(RealmLogLevel.error, "2");
-        await logTestMessage(RealmLogLevel.error, "2");
-
-        // turn off second isolate
-        Realm.logger.level = RealmLogLevel.off;
-
-        // log another message. second isoalte should not get it
-        await logTestMessage(RealmLogLevel.error, "first only");
+    final trace = Isolate.run(() async {
+      return (await Realm.logger.onRecord.first).message;
+    });
+    await Future<void>.delayed(const Duration(milliseconds: 100)); // yield
+    expectLater(trace, completion('Hey'));
+    Realm.logger.log(RealmLogLevel.trace, 'Hey');
+  });
 
-        // turn on second isolate
-        Realm.logger.level = RealmLogLevel.trace;
+  test('RealmLogger hookup logging', () async {
+    Realm.logger.setLogLevel(RealmLogLevel.off);
+    Realm.logger.setLogLevel(RealmLogLevel.all, category: RealmLogCategory.realm.sdk);
 
-        // log a another message
-        await logTestMessage(RealmLogLevel.trace, "3");
+    final logger = Logger.detached('Test');
+    Realm.logger.onRecord.forEach((r) => logger.log(r.level.level, r.message));
+    logger.level = Level.ALL;
 
-        // stop all isolates signal
-        await logTestMessage(RealmLogLevel.detail, "stop");
+    expectLater(logger.onRecord, emits(isA<LogRecord>().having((r) => r.level, 'level', Level.SEVERE).having((r) => r.message, 'message', 'error')));
 
-        return await completer2.future;
-      });
+    Realm.logger.log(RealmLogLevel.error, 'error');
+  });
 
-      await completer.future;
+  test('RealmLogger hookup hierarchical logging', () async {
+    Realm.logger.setLogLevel(RealmLogLevel.off);
+    Realm.logger.setLogLevel(RealmLogLevel.all, category: RealmLogCategory.realm.sdk);
 
-      return result;
-    });
+    hierarchicalLoggingEnabled = true;
+    Realm.logger.onRecord.forEach((r) => Logger(r.category.toString()).log(r.level.level, r.message));
+    Logger.root.level = Level.ALL;
 
-    final expected = [
-      const LoggedMessage(RealmLogLevel.error, "2"),
-      const LoggedMessage(RealmLogLevel.error, "2"),
-      const LoggedMessage(RealmLogLevel.error, "first only"),
-      const LoggedMessage(RealmLogLevel.trace, "3")
-    ];
+    expectLater(Logger('Realm').onRecord, emits(isA<LogRecord>().having((r) => r.level, 'level', Level.SEVERE).having((r) => r.message, 'message', 'error')));
+    expectLater(
```

Comment:
I'm not sure what's happening here - is the hierarchical logging something we get for free from the framework?